src/imscc/resource/html.ts

```diff
@@ -29,7 +30,11 @@ const templateHtml = (page: Page, id: string, options?: Config) => {
 <meta name="editing_roles" content="teachers"/>${styleTag}
 </head>
 <body>
-${page.content}
+${
+  typeof page.content === "string"
+    ? DOMPurify.sanitize(page.content)
+    : page.content
```

Comment:
When would it be that it's not a string?

Reply:
Above it defaults to ""

Reply:
https://github.com/OpenLearningNet/imscc-packager/blob/a5583dd678f5f217cd2eb9af98dea9f6a4595756/src/imscc/types.ts#L66

when it's an assessment

Reply:
.question for each section?

perhaps you can do:

```
const sanitizeAnswer = <T extends Answer>(answer: T) => ({
  ...answer,
  text: sanitize(answer.text)
});

const sanitizeAnswerArray = <T extends Answer>(answers: T[] | undefined) => answers ? answers.map(sanitizeAnswer) : undefined;

const sanitizeMatches = (matches: Match[] | undefined) => matches ?
  matches.map((match) => ({
    pair: [sanitizeAnswer(match.pair[0]), sanitizeAnswer(match.pair[1])]
  })): undefined;

const sanitize = (content: string | Section[]) => {
  if (typeof content === "string") {
    return DOMPurify.sanitize(content);
  } else {
    const sanitizedSection = content.map((section) => ({
      ...section,
      title: sanitize(section.title),
      question: sanitize(section.question),
      answers: sanitizeAnswerArray(section.answers),
      choices: sanitizeAnswerArray(section.choices),
      matches: sanitizeMatches(section.matches)
    }));
    return sanitizedSection;
  }
}

```

Reply:
Note: any user editable field that could possibly be reflected back to a user as HTML needs to be sanitized

Reply:
> do those sections also have HTML?

yes, only questions that can be in html format. currently i sanitized the question in coursemagic side because it's in escaped format, i sanitze it before escaping it. should the imscc-packger be the one that responsible for sanitizing the html?